src/pyseekdb/utils/embedding_functions/onnx_embedding_function.py

```
@@ -0,0 +1,392 @@
+        """
+        Initialize an ONNX embedding function.
+
+        Args:
+            model_name: Name of the model (used for cache directory naming).
+            hf_model_id: Hugging Face model ID.
+            dimension: Output embedding dimension.
+            download_path: Optional cache path override.
+            preferred_providers: Preferred ONNX runtime providers.
+        """
+        if not model_name:
+            raise ValueError("model_name must be a non-empty string")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)